lib/reservation_creator.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+class ReservationCreator
+  # Service Object to create reservations in the reservations controller
+  def initialize(cart:, current_user:, override: false, notes: '')
+    @current_user = current_user
+    @cart = cart
+    @cart_errors = cart.validate_all
+    @override = override
+    @notes = notes
+  end
+
+  def create!
+    return { result: nil, error: error } if error
+    reservation_transaction
+  end
+
+  def request?
+    !override && !cart_errors.blank?
+  end
+
+  private
+
+  attr_reader :cart, :current_user, :override, :notes, :cart_errors
+
+  def error
+    return 'requests disabled' if request? && AppConfig.check(:disable_requests)
+    return 'needs notes' if needs_notes?
+  end
+
+  def needs_notes?
+    (request? || override?) && notes.blank?
```

Comment:
Isn't `request? || override?` equivalent to `!cart_errors.blank?`, since `request?` is true if `override` is falsey and `override?` is true if `override` is truthy?